dlt/sources/sql_database/__init__.py

```diff
@@ -264,8 +272,15 @@ def sql_table(
     else:
         hints = {}
 
+    if primary_key:
```

Comment:
maybe add a comment here why this is treated differently than the merge key: because we need to overwrite what is found in the reflection.